presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoSession.java

```diff
@@ -549,14 +549,18 @@ else if (value instanceof Document) {
             List<TypeSignatureParameter> parameters = new ArrayList<>();
 
             for (String key : ((Document) value).keySet()) {
-                Optional<TypeSignature> fieldType = guessFieldType(((Document) value).get(key));
-                if (!fieldType.isPresent()) {
-                    return Optional.empty();
+                Object fieldValue = ((Document) value).get(key);
+                Optional<TypeSignature> fieldType = guessFieldType(fieldValue);
+                if (fieldType.isPresent()) {
+                    parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(key)), fieldType.get())));
                 }
-
-                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(key)), fieldType.get())));
+                else {
+                    log.info("Unable to guess document field %s type from class %s with value %s", key, fieldValue == null ? "null" : fieldValue.getClass().getName(), fieldValue);
+                }
```

Comment:
Logging only here looks a little weird in this method. We could put just `else` at the last of the  `instanceof` conditions if needed in the future. Could you revert some change and update like this?
```java
            for (String key : ((Document) value).keySet()) {
                Optional<TypeSignature> fieldType = guessFieldType(((Document) value).get(key));
                if (fieldType.isPresent()) {
                    parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(key)), fieldType.get())));
                }
            }
```

Reply:
Ok, removed